src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/tests/test_sf_commands.py

```diff
@@ -0,0 +1,87 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import unittest
+import os
+
+from azure.cli.testsdk import ScenarioTest, ResourceGroupPreparer
+from azure.cli.testsdk import JMESPathCheck as JMESPathCheckV2
+from azure.cli.testsdk.vcr_test_base import (ResourceGroupVCRTestBase,
+                                             JMESPathCheck, NoneCheck)
+
+class ServiceFabricNewClusterTest(ResourceGroupVCRTestBase):
+    def __init__(self, test_method):
+        super(ServiceFabricNewClusterTest, self).__init__(__file__, test_method, resource_group='cli-sf-test')
+        self.location = "westus"
+        self.resource_group = 'cli-sf-test'
+        self.cluster_name = self.resource_group
+
+    def test(self):
+        self.execute()
+
+    def body(self):
+        rg = self.resource_group
+        name = self.cluster_name
+        location = self.location
+
+        self.cmd('sf cluster new --resource-group {} --location {} --cluster-size 1 --certificate-subject-name test.com --vm-password User@1234567890'.format(rg,location), 
+                 checks=[JMESPathCheck('cluster.name', name)])
+
+
+class ServiceFabricGetAndUpdateTest(ResourceGroupVCRTestBase):
+    def __init__(self, test_method):
+        super(ServiceFabricTest, self).__init__(__file__, test_method, resource_group='cli-sf-test')
+        self.location = "westus"
+        self.resource_group = 'cli-sf-test'
+        self.cluster_name = self.resource_group
+
+    def test(self):
+        self.execute()
+
+    def body(self):
+        rg = self.resource_group
+        name = self.cluster_name
+        location = self.location
+
+        self.cmd('sf cluster list', checks=[JMESPathCheck('type(@)', 'array')])
+
+        self.cmd('sf cluster show --resource-group {} --name {}'.format(rg,name), 
+                 checks=[JMESPathCheck('name', name)])
+
+        self.cmd('sf cluster upgrade-type set --resource-group {} --cluster-name {} --upgrade-mode automatic'.format(rg,name), 
+                 checks=[JMESPathCheck('upgradeMode', 'Automatic')])
+
+        test_thumbprint = '9B609A389BD4597BEEFABFB363BF2BBF2E806001'
+        self.cmd('sf cluster client-certificate add --resource-group {} --name {}  --thumbprint {}'.format(rg,name,test_thumbprint), 
+                 checks=[JMESPathCheck('clientCertificateThumbprints[0].certificateThumbprint', test_thumbprint),
+                     JMESPathCheck('clientCertificateThumbprints[0].isAdmin', False)])
+
+        self.cmd('sf cluster client-certificate remove --resource-group {} --name {} --thumbprints {}'.format(rg,name,test_thumbprint), 
```

Comment:
This needs to test the multi-thumbprint scenario, as I have doubts that it will work the way you have things registered.